### ISSUE TITLE
16.12.1 unstable

### ISSUE THREAD

**Issue by JackalWorks** [TinkerGnome/Ultimaker2Marlin#56]
Prints seem to fail a lot more often with this than 16.08.2. The error message just says Error Stopped go to ultimaker... Not very descriptive, so I am not sure what is causing this. I am on a UM2E+
Now it also hanged while trying to read the list of files from the sd-card.

**Comment by ghost** [comment]
Hi, me too : 
with the new firmware (16.12.1) I got the same error "Error Stopped", this happen nearly everytime for long prints (for now, more than 1h), with stock SD card and Octoprint also. So I'll wait a correction because I double check my feeder, nozzle, temp and all in my 3D printer : I don't find any trouble.
I've an Ultimaker 2+ and a Bondtech extruder in 1.75mm (with 1.75mm Matchless Nozzle), I need the new firmware because I have to reverse the E axe to make the Bondtech working in the right way. With the stock extruder and the previous firmware (16.08.2) I never had this error before.
Please @TinkerGnome find what's going. But indeed, thank's for all your time developing this great potential firmware <3
![img_3943](https://cloud.githubusercontent.com/assets/6934787/21966748/a31b7c3c-db79-11e6-9bd8-02b6396ca0ea.jpeg)

**Comment by TinkerGnome** [comment]
At the moment I have not more information than - something non-reproducible goes wrong in your case - :worried:
Let's try to find out, what's different: which slicer? version? in case of Cura- any post-processing scripts? Issues only for specific gcode files? A different behavior, if the printer is re-started before every print?

**Comment by ghost** [comment]
I use only Simplify3D 3.1.1 (working without any trouble before this new firmware).
I always restart the printer before print, and also here because I need to (after the ERROR - STOPPED, we can't do anything at the screen, but in Octoprint it's working).
I've upload a zip with 2 gcode with the "ERROR - STOPPED" appear randomly, the config for Simplify3D and the .stl file.
[Error Stopped.zip](https://github.com/TinkerGnome/Ultimaker2Marlin/files/707678/Error.Stopped.zip)
I'll try with a new big 3D print file, but this one always bug on the new firmware, so you can try it to see the bug (I hope).

**Comment by JackalWorks** [comment]
I also use S3D 3.1.1. I did not try Cura. I downgraded back to 16.08.2. And it is rock stable again.

**Comment by unixhelden** [comment]
Tried both, Cura 2.3.1 and 2.4.Beta (and Beta2).

Prints with Octoprint via Serial or SD Card the same behavior. Sometimes the print passes somestimes this error appears, happend with the same gcode, first print success, next try error.

Stock Ultimaker 2+ nothing special.

**Comment by TinkerGnome** [comment]
> I downgraded back to 16.08.2. And it is rock stable again.

That's really odd, i have reports that are the exact opposite... Some users had weird issues with 16.08 and are happy again with 16.12... I have nothing changed that could be (directly) related to this. There's no starting point....

**Comment by JackalWorks** [comment]
I should point out that I have still other issues with 16.08.2 like the previously reported bugs with the UI like going back to the menu during print and hang in the UI. But when it comes to the actual print it completes successfully and never fails.

**Comment by Turtletrumpet** [comment]
There are some small problems with 16.08.2, but the prints works and don't stops by error.
The V16.12.x stops print after round about an hour with errors, that i haven't ever before:
![tinkerv16 12](https://cloud.githubusercontent.com/assets/21341682/22007619/722c9bbe-dc74-11e6-9624-a4d7be858753.JPG)
And that's the reason, why I use V16.08.2, untils 16.12.x is working. The stock firmware isn't an alternate for me ;). I love the tinker firmware!

**Comment by gr5** [comment]
tinkergnome maybe change the path planner lookahead from 16 points to 8 points to see if that helps?  Just for information gathering?

**Comment by andersolsson** [comment]
I also had issues (with the Tinker-MarlinUltimaker2go-HBK-dual-16.12.1) on the UM2Go om two occasions.
Two weeks ago I just found it like this after a print:
![2017-01-08-0246](https://cloud.githubusercontent.com/assets/7799379/22254605/5dc93574-e255-11e6-86d6-c32088d77fdd.jpg)
It worked fine next time I printed though, same file and same settings.

Today it suddenly started driving the head into the back of the printer in the middle of printing a robot. 
It continued pushing the head towards the back of the printer for maybe 5 seconds, then drove the head into the front of the printer for about one second, and then continued printing in the air at that spot.
I printed the file once again now though and then it worked fine.

I am thinking if it is some scheduling issue that overloads the PSU, or something like what gr5 mentions?

It might be that my case is related to EMI though, since the electrical connection for my printer lacks an earth wire at the moment. (Which can be felt if touching metal parts of the printer)

**Comment by gr5** [comment]
Maybe I'll reduce path planner by 2X and send out copies to whomever wants.  Just to collect some info.  Who will test it and what version of hardware do you need me to build?

**Comment by TinkerGnome** [comment]
Actually the tinker version does not use more RAM than the standard firmware, but more program memory ofc. I agree, that there is probably a buffer (or integer) overflow somewhere, but I'm afraid, that it's not in an obvious place...
None of these problems were ever reproducible with my machine/settings, there has to be a difference, but where...?
@gr5 Any encouragement is welcome.

**Comment by gr5** [comment]
>Actually the tinker version does not use more RAM than the standard firmware

Maybe it only uses more memory on the stack then.  I mean you must have at least one variable somewhere that isn't in the original marlin, right?  These strange crashes make me think "stack" as those errors can be very intermittent and very strange.

**Comment by derbroti** [comment]
It has to be something like writing to the wrong ram position, as none of the stop calls that would produce the error-stopped without further information, so it can not be related to a normal printing operation when older firmwares do not trigger anything - ok besides a very erratic safety jumper... (unlikely?)

For what it's worth I built a version for me that is printing the amount of free memory to the unused second USART interface - maybe it runs out of memory after all?

Otherwise happy hunting for bad pointers :|

**Comment by andersolsson** [comment]
I just realized one thing when looking at the failed print: It appears like it failed after completing a layer.
So it looks like both incidents on my machine happened when (or very close to) moving the Z-axis.

I have attached pictures and the gcode for the failed robot.
It is not the same gcode as the one where the printer stopped just after printing, that one is stuck on the SD-card at the moment.

Both robots have printed successfully when trying again with identical settings, the first one probably printed fine 30 times or so.
 
Another thing I realized: If it would make any sense, it is possible that my printer behaved strange for about the time of one layer, banging the head into the back and then the front, then starting to move normally again when it reached next layer. (Just based on the duration of the problem).

The first error, when the printer stopped after finishing the robot, was with default settings on the primary extruder by the way
The mid-print issue happened with the second extruder at 370 steps/mm, 1.75 mm diameter and 600mA extruder current.

[UM2G_41_infill_UltimakerRobot.zip](https://github.com/TinkerGnome/Ultimaker2Marlin/files/737517/UM2G_41_infill_UltimakerRobot.zip)
![2017-01-29-0507](https://cloud.githubusercontent.com/assets/7799379/22404174/f0d9ed20-e62b-11e6-95c7-9f1035fd72d9.jpg)
![2017-01-29-0508](https://cloud.githubusercontent.com/assets/7799379/22404173/f0d848ee-e62b-11e6-8c15-faf97883ba01.jpg)
![2017-01-29-0509](https://cloud.githubusercontent.com/assets/7799379/22404178/18ef6772-e62c-11e6-9cba-a0767f649d3b.jpg)

**Comment by ieol** [comment]
Beware the SD card. Sometimes seems major problems comes from the SD card.
Yesterday I got a failure printing a new project after 4 hours the print stop, the screen goes black, Led shut off and even heating of bed and hot end goes off I thought was the new Firmware installed a day before  (16.12.1) so I downgraded to 16.08.2 and restarted the print but... the job ended exactly in the same point and with the same problems (all shut off led,heating,display..) I tried to simulate a resume and the printer stop at the same point 74.70 mm.. I took out the sd card, made a scandisk but no problems comes out even if Scandisk sucks I know so... I made a copy of the entire SD Card and.. surprise! not all file were readable and the file I had in print take long to be copied... so after many tries I copied them all and formatted de SD Card (but not fast format, I made a complete instead) Copied again on the SD card and inserted on UM2, my print comes out perfect without any problems made a second print of 8 hours and NO problems comes out
So... take care that the SD card is in good condition before attribute any failure to the firmware 
Cheers to all and Thank again for this nice Firmware

**Comment by derbroti** [comment]
I can second the fact that a faulty SD card can lead to arbitrary head movements - but those errors would not result in a "error - stopped" message on the display?

**Comment by unixhelden** [comment]
And this error also happens when printing via OctoPrint (USB), so this is not SD Card related.

**Comment by Turtletrumpet** [comment]
I print only with octoprint and get the 'stop' error, too. Is it a good idea to make a factory reset, after flashed the V16.12.1? The older version i used is V16.08.2.

**Comment by ieol** [comment]
@derbroti a faulty or better a semi faulty or we should talk about a CRC RANDOM faulty SD card can lead at any kind of problem even a corruption of other data depending on the system with which it will interact (software, hardware combinations) it's not a simple problem when you got Random CRC errors (as the case of my SD card)
When I tried to read it trough a 7 doors USB HUB it was merely unreadable, so, I connected it to the frontal USB door and it go back to be readable (well with some retry during copy process)

@unixhelden @Turtletrumpet  Turtletrumpet indirectly gave the answer to you as the Octoprint have REFRESHABLE memory inside, this mean the same, (alike, similar) chip of a SD card... But take in account that SD card born to be erasable a number of time that is usually Bigger compared to a Flash Memory chip used for firmware (you can find more info on the datasheet of the flashmemory chip manufacturer)

BY THE WAY: I believe that TinkerGnome got really a lot of patience reading so many comments...
**SO thank you again for this Nice Firmware**

**Comment by ieol** [comment]
@andersolsson I see your comment now (posted 10 days ago) This is right what happened to my UM2 (the print head above the printing piece, all shut off except the head fan that it is not controlled by sofware (as it goes ON when you power the UM) and I am almost sure that it is an SD card problem. I suggest to you to copy all your files on a computer and format your SD card (don't make a fast format as it doesn't correct the problem)

**Comment by unixhelden** [comment]
@ieol nope, my octoprint stores and loads gcode and timelapses to and from a nfs share.
BTW, i never had any hickups with 16.08.2 or the original Ultimaker Firmware.

With 16.12.1 i get spontanous Errors like the one described above. Print stops, Nozzle in the Middle of the Print and Display states. "Error - Stopped". So i would rule out SD Card Problems. Same Gcode File runs perfect and the next Print is an Error.

pi@umocto:~/.octoprint $ grep 'mnt' config.yaml
  timelapse: /mnt/syno/Timelapse
  timelapse_tmp: /mnt/syno/Timelapse/UMtmp
  uploads: /mnt/syno/Uploads
  watched: /mnt/syno/Uploads

192.168.200.250:/volume1/OctoPIUploads/Ultimaker on /mnt/syno/Uploads type nfs 
192.168.200.250:/volume1/Timelapse on /mnt/syno/Timelapse type nfs 

I have 3 Octoprint Servers for 3 different Printers, all setup the same with this nfs share. And only this Firmware brings errors. If someone would provide a debug version of that firmware, i could make some tests with serial logging enabled.

**Comment by gr5** [comment]
@unixhelden - I have the build environment setup the same way UM does and tinkerGnome does.  I'd be happy to build a version for you but I don't know about this "debug version".  Is there a flag in configuration.h (sorry too lazy to check at t he moment).  If there is such a flag and then tell me about it and tell me what configuration I should build for (there's a dozen or so builds) and I'll make you one.

**Comment by ieol** [comment]
@unixhelden so let resume you use an even more susceptible connection mode for transferring gcode?  Really! seems right the correct way to test something new! And see if it is stable and error free...  gosh!

**Comment by gr5** [comment]
The fact that it works on 3 printers for years and "suddenly" stops working when simply upgrading the firmware is good enough for me.  It could be a coincidence with something else that happened but considering all the complaints I think there is a bug in this version of the code.  Probably not Tinker's bug - probably some bug that is only now showing up.  Maybe the interrupt service routine doesn't finish in time.  Maybe there is a stack error or initialization error or... something.

**Comment by ieol** [comment]
@gr5 there are no fact or the problem would already be solved... if you think that change in firmware is a simply upgrade you already write all... have a nice day

**Comment by unixhelden** [comment]
@ieol please. calm down.

i will summarize that for you: Multiple People experience the same issue when using 16.12.1. Multiple People say this error occurs with different slicers ( i personally tried s3d and cura (2.3 and 2.4beta1-2)).

All of them Report when they switch back to the older firmware this error does simply not happen at all. With the same hardware, the same slicer, the same sd card and or serial delivery via Octoprint. 

All of them say: We can print the same Gcode 2 Times, the first time it's ok, the second time the same gcode on the same sd-card or the same pipeline brings this error up.

Next i tell you from my point there is no sd card at all involved. I use this setup on THREE Printers without any issues, i run prints for my hub which are 36-72 hours long without any hickup or fault unless i run 16.12.1 on my um2+. 

@gr5 i'll come back to you. I have to run several orders till thursday on my um but i think there is a debug option allready in marlin, i will look into it.

**Comment by Turtletrumpet** [comment]
Clear & short: There is a problem in 16.12.1. In that Version the pint stops without useable error message. Nevertheless i love the Tinker"ware" ;). I hope the error will be found. I don't know, what i could do, to find the error ...
**@ieol** take a deep breath, be friendly and have a nice weekend ;)

**Comment by ieol** [comment]
@unixhelden @Turtletrumpet LOL
I never write 16.12.1 is flawless, I just point out the way you are all begging for a flawless firmware but just write down information that ARE INCONSISTENT for who is developing it!
(UPPERCASE is just for underlining not shouting... I prefer instead of **uppercase**)

oh @Turtletrumpet have a nice weekend too ;)

**Comment by Turtletrumpet** [comment]
@ieol please start first and write consistent messages, who helps to find the error. At the moment you keep writing, that our feedbacks ARE INCONSISTENT. My opinion: wrong way to solve problems.

**Comment by Turtletrumpet** [comment]
the art of programming is to ask the right questions at the right moment and made a progam of that.
Just after the second world war, when i start programming, my programms stops with debug information ;).

**Comment by ieol** [comment]
@Turtletrumpet **LOL** are you silly? or too tired?

**Comment by Turtletrumpet** [comment]
@ieol ask this question first yourself ....

**Comment by gr5** [comment]
When reading a comment that appears angry at first one should assume the writer was laughing or smiling when he wrote it and was trying to be friendly.  There's usually at least 2 ways to read every post - always assume the friendlier version as that is usually the truth anyway.  I suspect ieol was not angry when he made any of these posts and didn't mean to hurt anyone's feelings.  @ieol I suggest you fill all of your posts with smiley faces in the future :)  All caps doesn't always mean shouting.  It could be mild emphasis. :)  Of course there is a third possibility - that you are dealing with a fool and fools should be ignored.  But always assume the best possibility!

**Comment by TinkerGnome** [comment]
back to topic... some thoughts:

- for 16.03 and 16.08 a more recent version of the Arduino sd-card library was used and multiple users had problems with read errors from the sd-card
- at least two of them solved this by re-routing and/or swapping of the two ribbon cables between the mainboard and the controller board at the front of the printer
- to avoid this issue i (tentatively) switched back to the former sd-card library and got no new reports about "printing out of print area"
- now with 16.12 some other users have different issues, while for all others (including me) all firmware versions are working just fine
- the new problem is unrelated to the sd-card, it doesn't matter how the printer is feeded with gcodes
- the un-answered question is: where is the difference? hardware? differences between the mainboard revisions? interfering signals (because the buildplate heater is in PID mode)? user workflow? weak connections? overheating (do we ask too much of the poor old Arduino)?
- ok, this are multiple questions... 😃 

AFAIK the OLED display (and the encoder wheel) is also connected via the ribbon cables, communicates with the mainboard through I2C protocol and the whole process is handled inside an interrupt service routine. Can a bad signal on the I2C bus screw up the Arduino in a way that it reads bad values from input pins anywhere else? I really don't know.

@gr5 - it can still be a stack overflow, but i doubt it. This should be reproducible, shouldn't it? If the printer is powered on, the memory state is always the same. If I print the same file again, it should fail at the very same point. I usually start several prints in a row, long prints, short prints again and again and the printer is running all the time. If there where a serious memory overflow - I'm sure that I would have seen the problem too...

> the art of programming is to ask the right questions at the right moment

So the question is - "what is the right question?"... 😃

**Comment by gr5** [comment]
One thought - in "normal" Marlin there aren't quite so many things being updated on the screen.  Maybe if you update one thing too many then it takes a bit too long and some timing problem occurs - like on windows when one process is using too much cpu time.  I haven't looked at the display code but maybe the I2C ISR is just pumping out data from some buffer to the screen.  Maybe the buffer for that is overflowing?  Is there a buffer for that?  Maybe the ISR routine takes a different amount of time and sometimes it takes too long?  Maybe the difference is some subtle update - maybe a test is to display FEWER things on the "printing" screen - or to go into the tune menu and leave it there and see if it still crashes?  Does this even make sense?  I don't know without looking at the code.

**Comment by derbroti** [comment]
@gr5 but from 16.08 to 16.12 the display functions did not increase in amount, right?

When comparing the the releases, there are not many noticeable differences (when ignoring the SD reader change) but one major looking switcharoo and change is in stepper.cpp in the ISR 
```
// "The Stepper Driver Interrupt" [...]
ISR(TIMER1_COMPA_vect) [...]
```

in Marlin_main.cpp there are quite a few memsets which make the main loop a little slower but that shouldn't be a problem...

It nevertheless can still be the rolled back SD library - maybe it has a weird bug in it? 
@TinkerGnome I didn't look into it: where does the "old" (the one you are using now) and "new" (the previously used one) SD library come from?

@all are you people using the prebuild *.hex files from tinker or are you compiling yourselves? 
-> If so, which Arduino IDE version are you using?

**Comment by gr5** [comment]
I know that TG uses arduino compiler 1.6.10.  TG helped me set up the build environment similar to his and similar to how Ultimaker does the build.  I took lots of notes if you want my notes.  I was going to post them on the forum but forgot so I just did it a minute ago:
https://ultimaker.com/en/community/38154-how-to-build-marlin-the-way-tinkerghome-and-ultimaker-do-it-if-you-have-windows

**Comment by gr5** [comment]
I'm happy to build someone a version with the bang-bang heated bed if they want to test to see if that was causing their problem.  It would be nice to start eliminating things.  You could just air-print with a cold nozzle.  Although it might take all 4 axes moving to cause the crash.

**Comment by derbroti** [comment]
@gr5 Interesting, a few month ago he was using 1.6.5 and the package script to this day still defines the arduino env as 165
I haven't looked into how much of the default arduino lib stuff is used in the firmware but it is certainly a point that changed

Oh and testing without actually wasting filament is a good idea - when you run the print with unloaded filament you would also run the 4th axis just as normal - but you might want to either unscrew the nozzle or set the temperatur to some low temp and disable the cold-extrusion prevention

and one last thought: pid-bed was also there in TG older firmware, that apparently runs fine, so I can not imagine that that is the problem - but who knows... timing stuff? 
I haven't had time to check with the old firmware but with the new 16.12.1 my LEDs start to flicker when printing (no, not a power issue) but the PWM / timer stuff is too slow but only when printing

**Comment by gr5** [comment]
The lights shouldn't flicker "when printing" but when "up to temperature".  The PID controller says "power at 100%" when it's 10C cooler than goal temperature.  But when it gets within 10C then the lights start to flicker.

**Comment by gr5** [comment]
On December 30, 2016 TG said he was using 1.6.10 in an email to me.

**Comment by derbroti** [comment]
No, mine also flicker (but almost not noticably) after heating up and at constant temperatures.

and here https://github.com/TinkerGnome/Ultimaker2Marlin/issues/29

**Comment by gr5** [comment]
Okay maybe I found something.  This line of code was pulled out on december 26 from stepper.cpp (last line in code chunk below).  TG - I don't understand this exactly but I think setting busy=true may be important.  This isr is like a thread and can happen at any moment and the motion planner needs to know that this block is now being sent to the steppers so the planner doesn't change it mid-stride.  Right?  Could this be it?

      // Anything in the buffer?
      current_block = plan_get_current_block();
      if (current_block != NULL) {
       current_block->busy = true;               // this line removed

**Comment by derbroti** [comment]
@gr5 Sadly no, because the block that is returned by `plan_get_current_block();` is set to `block->busy = true;` automatically (see [planer.h:133)](https://github.com/TinkerGnome/Ultimaker2Marlin/blob/geek_mode/Marlin/planner.h#L133)

**Comment by TinkerGnome** [comment]
If anyone is brave enough 😄 to try different compiler versions - the instructions from gr5 are great - it's not that complicated. Any version from 1.0.3 on should be able to compile the code.
BTW: Yesterday I installed on my machine  a version that was compiled with Arduino 1.8.1. There's no noticeable difference in behavior so far. But who knows... if you want to rule this out - give it a try!

**Comment by qwerty8224** [comment]
Cura 15.04.6 errors no 3 day printing

**Comment by derbroti** [comment]
@qwerty8224 Which firmware? Which printer?

**Comment by qwerty8224** [comment]
Tinker-MarlinUltimaker2-dual-16.12.1.hex
UM2
Cura 2.4beta2 Error

Included Neatbed PID got an error cura 15.04.6

**Comment by rolfx** [comment]
"Tried printing out of printing area"...

from november 2016 - after mounting the bondtech feeder on my UM II ext - I used Tinker-MarlinUltimaker2-16.08.2.hex. problem-free.
building an object of z-height 300 3 days ago the UM stopps at Z 230mm with the error "tried printing out of printing area." 
using youngest firmware "Tinker-MarlinUltimaker2extended-16.12.1.hex" the result is the same: UM stopps ptnting at Z 230mm.
on an tipp from martin bondeus - bondtech owner - I checked the UM display and find adjust z-position. the nozzle goes to Z 230mm and prompt me to move the platform to the nozzle. one can move the platform about 50 mm - then it stopps.
finding today your discussions over the issues #56 I find one sentence that I do not understand: "to avoid this issue i (tentatively) switched back to the former sd-card library and got no new reports about "printing out of print area".
can _I_ switch back?    to what?? 
I am not coder at all!
and: which is _the_ stable version to build parts up to the max height of the UM II xt?

one sentence to the theme "error bits at copying to SDcard". as I know all "write and read"-actions are verified by the system(s) by "Check totals". otherwise we would have lots of mistakes in pictures and movies on/from SDcards.

thanks for your patience

**Comment by TinkerGnome** [comment]
@rolfx - Your problem is perhaps easier to solve. It sounds like you had installed a wrong variant before (not the "extended"). You should check the stored maximum z-position:
Advanced -> Preferences -> Print area
Change the max. Z position to 325.0 and re-adjust the z-position afterwards. This should do the trick (in your case).

**Comment by rolfx** [comment]
indeed - good tipp. it works. on the other hand: I wrote and I installed the _extended_ firmware - at least the one that i declared as it. 
thanks!

**Comment by rolfx** [comment]
one questions: is there a documentation about all menu-points of the tinker-marlin firmware?

**Comment by gr5** [comment]
You mean you want like a visual tree view of all the menus?

**Comment by rolfx** [comment]
yes - it seems easier to me to get an overview - and a print in the drawer ;-)

**Comment by TinkerGnome** [comment]
Meanwhile I have a strong suspicion about the reason for instability, summarized in: short voltage drops due to an overloaded power supply...
Thanks to @gr5 you can check this theory now. I published a new (testing) release right now with his smart power limiter.
If you have one of the issues described earlier: feel free to test [V17.02.1](https://github.com/TinkerGnome/Ultimaker2Marlin/releases/tag/V17.02.1). Please use the default values for the power budget in the beginning and report back, if it improves the stability.
👍 ❓ 👎

**Comment by derbroti** [comment]
Did anyone measure that? My printer stays at comfortable 100W during print.

The only possibility: When you try to print abs in a freezing/windy room with >=100C bed temp in bang bang mode and a >=40W heater. 

Only during heatup of bed and nozzle (50W heater limited to ~30W) I do indeed reach around 220W (Input power not output)

**Comment by EldRick** [comment]
I completed an 18-hour print with no error-stopped, using 17.02.1 This on UM2, Bondtech, 35w heater.

**Comment by gr5** [comment]
But you never had a problem before either, right?

**Comment by EldRick** [comment]
Not before 16.2.1

**Comment by qwerty8224** [comment]
7 days no printing errors )))

**Comment by foreachthing** [comment]
>5 days no printing error - until yesterday. Long job (>20h) stopped after 3 hours. :-\ Sorry, no further info.

**Comment by gr5** [comment]
Just want to know if the screen was locked up or if the abort menu option was available but the printer had stopped or if it had rebooted and was doing what you would expect if there was a power glitch (back at main screen).

**Comment by qwerty8224** [comment]
day 17 no printing errors )))

**Comment by gr5** [comment]
@qwerty8224  great news but which version?  17.02?

**Comment by qwerty8224** [comment]
17.02.1

**Comment by qwerty8224** [comment]
Used Dual Extruder Firmware (reprap/marlin) and Single Extruder Firmware (UltiGcode) 
NO ERRORS )))

**Comment by SKlasen** [comment]
Hi guys, here are my experiences with 17.02, 16.08 & 16.12 on an 3 year old UM2 converted to Cremas Dual Extrusion Kit with watercooling, 2x Bondtech QR Feeder, 2x 35W heater and Olsson Block MK3 with 0.6mm matchless nozzles.
Most time I´m printing bigger parts like induction parts from ABS for automotive development which take ~30 hours each.
After the dual conversion I startet with 16.12. First 2-3 prints where ok, than then printer stopped after ~20h with the error message. I had to switch the printer off and on to get access to the menue again.
Then I switched back to 16.08. Long prints didn´t stop, but I had two prints where the extruder stopped working. I aborded the prints and tried to manually move material with the jogwheel through the menue. No reaction by the feeder! I switched the UM2 off and on and everything was ok again.
Then I tried 17.02. Some long prints where ok, but tonight the printer stopped again after ~20h with an error message and no more reaction. So I had to switch it off and on again.
The part was the same as printed with 16.08 and 16.12, but I used different GCode and the error apeared at a different Z-height.
Today I flashed the original Marlin Firmware from Cura 2.4 onto the printer, did a factory reset, changed the E-steps for the Bondtech and started a new print with the same GCode as used before with 17.02 firmware. I will let you know in a few days about the results. Maybe the firmware is not my only problem but the 3 year old mainboard?

Regards, Stephan

**Comment by gr5** [comment]
I'd like you to try 16.01 if not too much trouble.  I think that's the most recent stable version but it's really hard to tell when it takes 50 hours of printing to find out.  One can't exactly just change one line of code at a time, then print for 50 hours to see if it's stable.  My theory is we are using more and more memory with newer and newer versions and either heap memory or stack memory fills up.  But it's just one possible theory.  It could indeed be your hardware.  If I could dedicate 8 hours to the problem (I can't) I'd try to get codeblocks to run a simulated print and look at the stack and heap to see how close we come to having memory fill up.  This could be a total waste of time however.

**Comment by SKlasen** [comment]
Let me first try some prints with OE firmware during the next days, then we know 100% if there is an additional hardware problem on my printer or not. I have to do minimum 5 prints which take ~30h each.
After that (if there where no hardware problems) I will try the same with 16.01.

The problem first appeared after changing to dual extruder in combination with the Tinker FW.
Before I never knew about that firmware (sorry :-) ) and printed 3 years with Marlin without any of these issues at all. But I like the Tinker so much because of the settings and the dual feature, so we MUST get these problems solved :)

**Comment by gr5** [comment]
Thank you, yes I agree with all your points and I'm sure TinkerGnome agrees as well.

**Comment by qwerty8224** [comment]
Printed 15 hours all is well))) maybe 30 it's a lot and yet overflow.

SD card reads and the head does not travel around))))

**Comment by gr5** [comment]
@qwerty8224 - I don't understand what you are saying.  Are you just summarizing what @SKlasen said?  Or do you have new results?  Are you saying it works for you for 15 hours?  And that you will try 30 hours later?  I don't understand.  Also which version?

**Comment by qwerty8224** [comment]
I do not speak English sorry (((
Yes, 15 hours printed  worked well.
18 days NO ERRORS 17.02.1

**Comment by TinkerGnome** [comment]
@gr5 Stephan gets [this error message](https://ultimaker.com/en/community/17656-ultimaker-2-dual-extruder-einstellen?page=25#reply-174319) after several hours of printing.

**Comment by gr5** [comment]
Well that sucks.  So it seems like the bug is still there in version 17.02.

**Comment by SKlasen** [comment]
Yes the bug is still there in 17.02. I also mentioned that there is a ~5mm blob of material where the print stopped, so it seems that the x- and y- movement stopped but it was still extruding for some seconds. After that the printhead had moved into it´s zero-axis corner.

**Comment by Turtletrumpet** [comment]
@gr5 Does the Tinker V16.01 works with octoprint?

**Comment by SKlasen** [comment]
Ok, after 5 days nonstop printing with Marlin 2.4 no problems at all. I finished 4 long prints and the 5th will be finished in the morning. So I have no hardwareproblem with my motherboard!
But all Version from 16.08 and later caused bugs as described before.
So what's next to try? Which is the latest stable version? 16.01?

**Comment by gr5** [comment]
16.01 should be fine.  That's the most recent version that I trust to be stable.

**Comment by SKlasen** [comment]
Ok, but to move forward with this issue: Which Version should I test? Maybe we have to go back to the last stable version an make a new release based on the last stable version? I can print every day always the same big object to test a new release.

**Comment by gr5** [comment]
Unfortunately a year's worth of changes have been added since the last stable version.  It would really suck to have to do that.  Instead it would be best if we had a testable theory as to the cause.  One theory was that it had to do with power distribution code.  That code was improved in version 17.02 only.  Another theory is that the machine is running out of stack space memory.  That is difficult to test.  I think there is a way to blast debug info out the USB port including free memory.  We could see if that gets very low during a long print.  I'm not sure how to do all that.  Also it should be able to simulate a long print in codeblocks and dump free memory to a log file there also.   Another theory is that there is some timing issue (e.g. interrupt service routine is entered a second time before it completes the first time?).  Or maybe a null pointer bug.  I think stack overflow is most likely.

**Comment by TinkerGnome** [comment]
>  (e.g. interrupt service routine is entered a second time before it completes the first time?)

AFAIK all interrupts are disabled while an ISR is executed. I think this reason can be ruled out.

**Comment by gr5** [comment]
Yeah I agree.  Stupid idea.  But if it runs too much the planner has no time to run.

Someone should try going here:
maintenance -> advanced -> preferences -> user mode
and disable geek mode to see if that makes any difference.  Maybe something in there is using too much cpu time or too much stack space.

**Comment by goopyplastic** [comment]
Anecdotally but I built up a clone and had problems where the head would roam around randomly during a print then get a stopped error. I ended up putting aluminum ducting tape around the lcd ribbons, tape on the outside of the lcd housing, and a ferrite around the lcd ribbons (I also went crazy and put ferrites on all the motors) ... and the problem went away. I also had a similar issue with a makergear m2 that would do this when using an external lcd / sd card. The wanhao duplicator 6, a blatant clone of the ultimaker also has aluminum ducting tape around the lcd cables stock, maybe this is why. It's cheap to try.

**Comment by Turtletrumpet** [comment]
Very interesting. I will try this tomorrow. Thanks!

**Comment by derbroti** [comment]
This might be a solution for some... and I can second that: a bad connection (interference, too long cable, loose connection, etc.) can lead to thos erratic problems but for example SKlasen reported that the problems went away with the original and old firmware - and that is too much of a coincidence that precisely at that time the possible bad connection to the SD card "fixed" itself

**Comment by gr5** [comment]
If you are getting sporadic movements where the head suddenly moves to the front or left side of the print area and then continues where it left off.  Or if you get "print out of area" or something like that.  Then it's probably a hardware issue reading the SD card.  I had a problem with dust and a hair on mine.  

But if the printer is crashing and/or locking up the display then it might be software.  They may seem like similar symptoms but they are quite different.

**Comment by SKlasen** [comment]
Printed another week with OE Firmware and no issue at all. So it is 100% firmware related issue.
Someone mentioned to switch off Geek mode. What is that for?

**Comment by gr5** [comment]
When printing, in geek mode, it shows you all kinds of things like temperatures, z position, feedrate, flow rate, time left and so on.  In "normal" mode it only shows you I believe time left.  I have this (desperate?) theory that geek mode is causing the crashes somehow.

**Comment by SKlasen** [comment]
I printed 5 more days my long prints with OE firmware without any issue. From now on I will try 17.02 again and disable Geek mode!

**Comment by gr5** [comment]
MORE DATA!  I have a customer, I think he is "limitedRacing" maybe on the forums.  He has the medusa feeder and 16.01 and had a screen freezes but it would finish printing.  He couldn't go into tune menu but not a disaster.  I suggested he turn off "geek mode" and he has printed 30 hours since then without any screen freezes (tune menu still works).  I also had a screen freeze recently on 16.01 (the first I've noticed) and have been doing non geek mode.  But it's rare that I pay that much attention the screen after the first few layers are done.

**Comment by andersolsson** [comment]
I just realized today that the first UM2 that I bought has been running on 15.12 for quite some time now and I never had any issues on that one. I always used geekmode on it and have been printing similar models compared to the one that made my UM2Go freeze with 16.12.1.

So if 16.01 has screen freezing issues, maybe one can find a clue in the changes made when 15.12 became 16.01 ?

**Comment by gr5** [comment]
I've been using 16.01 for a year and only noticed screen freezing once.  I think maybe it's not noticeable unless you actually look at the screen.

The problems with 16.01 seem pretty minor compared to later versions which can crash many hours into a long print.  Is it possible you had a few screen freezes over the last 18 months Anders but just didn't notice?

**Comment by musyne** [comment]
Is 16.01 compatible with the + models? (UM2+ in my case)
I see it's listed from 16.03.01 and above.
I was using 16.08 (I think) but I reverted to the default firmware due to the crash.

**Comment by andersolsson** [comment]
Sorry for my late reply gr5.
It is possible or even likely that I did not notice of course, if screen freezing it is that rare.
However I have been keeping a closer eye on the display lately when printing boron carbide (as I have to be there to monitor those prints anyway), and I haven't noticed anything strange.

**Comment by gr5** [comment]
@musyne 16.01 rotates the extruder the wrong way.  Personally I would probably install 16.01 for UM2 and then swap two pins on the extruder stepper.  But I've done that a few times and may be too scary if you've never swapped pins before.  Takes 10 seconds for an expert.  Also you have to double steps/mm but with tinkerware that's trivial - it's a menu item.

**Comment by musyne** [comment]
@gr5 yeah I've touched the board before but I'd rather not switch the pins for a temporary solution. Thanks for the advice I'll try another release and keep you posted if it works.

**Comment by musyne** [comment]
@gr5 Do you think you can add support for the + to 16.01 branch since it seems to be the last stable version?
I don't how much work it would be.
I'm back to the default firmware since a while now and I really miss TinkerGnome.

**Comment by SKlasen** [comment]
Any updates here? I´m using OE firmware for almost half a year now but want to install tinker again.
So 16.01 is the last stable version. Can we modify 16.01 so the stepper works as it should?

**Comment by SKlasen** [comment]
A few years later using Marlin, I´m testing 16.01 again with long prints.
